gkraken,nixos/gkraken: Drop

OPNA2608 opened the pull request:
It was [switched to maintenance mode on 2020-12-09](https://gitlab.com/leinardi/gkraken/-/releases/0.14.5), and [marked deprecated in favour of Coolero/CoolerControl on 2022-02-23](https://gitlab.com/leinardi/gkraken/-/releases/1.2.0).

We have [CoolerControl packaged since 24.05](https://github.com/NixOS/nixpkgs/commit/c981cb00dd1a78970addf91aa8c381a8c45fddb4), so we can now drop gkraken.

CC @codifryed (upstream maintainer of both projects, and package maintainer in Nixpkgs of the latter)

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Aleksanaa commented on nixos/modules/hardware/gkraken.nix:
There's also a module alias in `nixos/modules/rename.nix`. Would be better to add that as well.

OPNA2608 replied:
TIL! Added an entry for it there.

Aleksanaa commented on nixos/modules/rename.nix:
```diff
@@ -33,6 +33,7 @@ in
       systemd-logind API). Instead of using the module you can now
       simply add the brightnessctl package to environment.systemPackages.
     '')
+    (mkRemovedOptionModule [ "hardware" "gkraken" "enable" ] "gkraken was deprecated by coolercontrol and thus removed from nixpkgs.")
```
```suggestion
    (mkRemovedOptionModule [ "hardware" "gkraken" "enable" ] ''
      gkraken was deprecated by coolercontrol and thus removed from nixpkgs.
      Consider using programs.coolercontrol instead.
     '')
```

OPNA2608 committed:
gkraken,nixos/gkraken: Drop

It was switched to maintenance mode on 2020-12-09[1], and marked deprecated in favour of Coolero/CoolerControl on 2022-02-23[2].

We have CoolerControl packaged since 24.05[3], so we can now drop gkraken.

[1]: https://gitlab.com/leinardi/gkraken/-/releases/0.14.5
[2]: https://gitlab.com/leinardi/gkraken/-/releases/1.2.0
[3]: https://github.com/NixOS/nixpkgs/commit/c981cb00dd1a78970addf91aa8c381a8c45fddb4
